common/locale.go

```diff
@@ -0,0 +1,38 @@
+package common
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+// LocaleCode the locale code to use to determine the language to use
+const LocaleCode = ContextKey("LocaleCode")
+
+// SetLocaleCode sets the Locale code used to set the language
+func SetLocaleCode(req *http.Request) context.Context {
+	lang := ExtractLangFromSubDomain(req)
+
+	if c, err := req.Cookie("lang"); err == nil && len(c.Value) > 0 {
+		lang = ExtractLangFromCookie(c)
+	}
+
+	return context.WithValue(req.Context(), LocaleCode, lang)
+}
+
+// ExtractLangFromSubDomain returns a language based on subdomain
+func ExtractLangFromSubDomain(req *http.Request) string {
+	lang := "en"
+	if strings.Split(req.Host, ".")[0] == "cy" {
+		lang = "cy"
+	} else {
+		lang = "en"
+	}
+	return lang
```

Comment:
You could simplify this:
```
if strings.Split(req.Host, ".")[0] == "cy" {
    return "cy"
}
return "en"
```
Also I know its unlikely but _potentially_ `strings.split` might return nil or empty which would cause `[0]` to nil pointer. Consider adding a defensive check:
```
args := strings.Split(req.Host, ".")
if len(args) == 0 {
   // log defaulting to en
    return "en"
}

Reply:
Ah yes, I forget this is a go best practice thing, multiple returns in a function, I have built up a habit of avoiding multiple returns in a function as then logic flow can be directed by many returns of a function and can be hard to know when debugging that function x actually breaks out given the many situations it can be called. 

Will conform to Go best practices though :D, so thanks for pointing this out.

Also added the arguments check.